Sources/MacroTesting/AssertMacro.swift

```diff
@@ -602,7 +602,9 @@ private class FixItApplier: SyntaxRewriter {
               .location(for: oldNode.position, anchoredAt: oldNode, fileName: "")
               .offset
             if node.position.utf8Offset == offset {
-              return newNode
+              if node.kind == oldNode.kind {
+                return newNode
+              }
```

Comment:
This addresses an issue that I encountered while applying fix-its via this assert macro. In this fix-it I remove an attribute (attached macro) from a type declaration where the attribute is the only one present. The fix-it I create has a `change` that replaces the type's `AttributeListSyntax` (containing the attribute) with an empty one.

The issue arose because this code only checked for the offset of the nodes it's iterating over, not accounting for the fact that multiple nested nodes can share the same offset. For example, all of the following node have the *exact* same offset:

```swift
- CodeBlockItemListSyntax:
  ├─ [0]: CodeBlockItemSyntax:
           ├─ EnumDeclSyntax:
                   ├─ AttributeListSyntax:
                           ├─ AttributeSyntax:
...
```

This meant that when the fix it was applied to swap out the `AttributeListSyntax` for an empty one, the `CodeBlockItemSyntax` (or it's parent list if nothing preceded this declaration) was exchanged for an empty `AttributeListSyntax` due to this matching logic. 

The update simply adds an additional check to ensure that the syntax kinds match before performing the replacement.

Reply:
I've just take a quick look at this PR (so I'm not sure) but isn't it this issue: https://github.com/apple/swift-syntax/pull/2337 ?

Reply:
It looks like that PR may be similar, but this fix in particular is related to the `FixItApplier` class created in this file, which appears to differ from the fix in the referenced PR

Reply:
If I'm not mistaken, @stephencelis and @mbrandonw have taken `FixItApplier` from the `swift-syntax` repo. So, I guess it would be a good idea to update the entire syntax rewriter in the same way it has evolved in the original repository.

Reply:
I checked out the `FixItApplier` in the `swift-syntax` repo, but it doesn't appear to be related: https://github.com/apple/swift-syntax/blob/23d207d0b9057f02fcbdb2813f6ec832514c457c/Sources/_SwiftSyntaxTestSupport/FixItApplier.swift#L16

@stephencelis @mbrandonw Am I missing something in relation to this type?

Reply:
Because it was refactored and improved. Here is one of the PRs containing changes to it: https://github.com/apple/swift-syntax/pull/2226/files#diff-14e78b1c33d017a6ab9041f5fb8274b8db9f22e0a090abca326cd567199e9e3d.

Reply:
Ahhhh, I see. Could this be updated in a followup PR or do you feel that it'd be better to do it as part of these changes?

Reply:
Personally I would recommend fallowing @stephencelis led and:

> (...) this is really an upstream issue that will hopefully be fixed by Apple soon. Because of that, I'm inclined to think we should limit this PR to just fixing the formatting issue to minimize future churn when reintegrating code back from Apple in the future.

limit the PR to just formatting issue and when we will close this PR https://github.com/apple/swift-syntax/pull/2337 in `swift-syntax` I would recommend incorporate the new version of `FixItApplier` to this repo.

Reply:
@SomeRandomiOSDev I think a second PR would be more appropriate. Want to fix up this PR to just the formatting change and then do a follow-up with the new upstream code?

Reply:
Works for me! :)

Reply:
Latest updates further trim this PR down to just the formatting issues